gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsService.java

```diff
@@ -0,0 +1,508 @@
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * The sorting keys for listing change requests. The only currently supported sorting key is the
+   * change sequence.
+   */
+  enum ChangeRequestSortingKey {
+    CHANGE_SEQUENCE("changeSequence");
+
+    private final String selector;
+
+    ChangeRequestSortingKey(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+  }
+
+  /**
+   * The sorting order for listing change requests.
+   */
+  enum ChangeRequestSortingOrder {
+    DESCENDING, ASCENDING;
+
+    public String selector() {
+      return this.name().toLowerCase();
+    }
+  }
+
+  /**
+   * Class that for specifying DNS record options.
+   */
+  class DnsRecordOptions extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 201601261646L;
+
+    DnsRecordOptions(DnsServiceRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the DNS record's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all record fields are returned. {@code
+     * DnsRecordField.fields} can be used to specify only the fields of interest. The name of the
+     * DNS record always returned, even if not specified. {@link DnsRecordField} provides a list of
+     * fields that can be used.
+     */
+    public static DnsRecordOptions fields(DnsRecordField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("rrsets(").append(DnsRecordField.selector(fields)).append(")");
+      return new DnsRecordOptions(DnsServiceRpc.Option.FIELDS, builder.toString());
+    }
+
+    /**
+     * Returns an option to specify a page token.
+     *
+     * <p>The page token (returned from a previous call to list) indicates from where listing should
+     * continue.
+     */
+    public static DnsRecordOptions pageToken(String pageToken) {
+      return new DnsRecordOptions(DnsServiceRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * The maximum number of DNS records to return per RPC.
+     *
+     * <p>The server can return fewer records than requested. When there are more results than the
+     * page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static DnsRecordOptions pageSize(int pageSize) {
+      return new DnsRecordOptions(DnsServiceRpc.Option.PAGE_SIZE, pageSize);
+    }
+
+    /**
+     * Restricts the list to return only zones with this fully qualified domain name.
+     */
+    public static DnsRecordOptions dnsName(String dnsName) {
+      return new DnsRecordOptions(DnsServiceRpc.Option.DNS_NAME, dnsName);
+    }
+  }
+
+  /**
+   * Class for specifying zone field options.
+   */
+  class ZoneFieldOptions extends AbstractOption implements Serializable {
```

Comment:
See earlier comment.
1) I think we usually suffix with `Option`
2) We typically create one per method or set of method that are related (and get the same) and use a common
verb. like get, read, write, list...